src/Geocoder.php

```diff
@@ -8,216 +8,252 @@
-    // - option AutoSuggestOption::fallback_language(): For normal text input, specifies a fallback language, which will help guide AutoSuggest if the input is particularly messy. If specified, this parameter must be a supported 3 word address language as an ISO 639-1 2 letter code. For voice input (see voice section), language must always be specified.
-    public function autosuggest($input, $options=array())
-        {
-        $parameters = array("input"=>$input);
-
-        foreach ($options as $option)
-            $parameters = array_merge($parameters, $option);
-
-        return $this->performRequest("autosuggest", $parameters);
-        }
-
-
-    ////////////////////////////////////////////////////////////////
-
-    // Prepare the call to the API server
-    private function performRequest($command, $parameters)
-        {
-        // add the apikey to the parameters
-        $parameters["key"] = $this->api_key;
-
-        // make an array out of the dictionary so that each element is a key value pair glued together with an '=', and urlencode the parameters
-        $param_array = array();
-        foreach ($parameters as $key=>$value)
-            $param_array[] = "$key=" . urlencode($value);
-
-        // glue the array together unto a string with elements connected with '&'
-        $params = implode("&", $param_array);
-
-        // put the whole URL together now
-        $url = $this->base_url . $command . "?" . $params;
-
-        // cal the server
-        $json = $this->call($url);
-
-        $data = json_decode($json, true);
-
-        if (isset($data["error"]))
-          {
-          $this->error["code"]    = $data["error"]["code"];
-          $this->error["message"] = $data["error"]["message"];
-          $data = false;
-          }
-
-        return $data;
-        }
-
-
-    // Make the call to the API server
-    private function call($url)
-        {
-        $handle = curl_init();
-
-        // set the options
-        curl_setopt($handle, CURLOPT_URL, $url);
-        curl_setopt($handle, CURLOPT_HTTPHEADER, array("X-W3W-Wrapper: " . $this->header));
-        curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($handle, CURLOPT_ENCODING, "");
-        curl_setopt($handle, CURLOPT_MAXREDIRS, 10);
-        curl_setopt($handle, CURLOPT_TIMEOUT, 30);
-        curl_setopt($handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
-        curl_setopt($handle, CURLOPT_CUSTOMREQUEST, "GET");
-
-        // make the call
-        $output = curl_exec($handle);
-        if ($output == false)
-            {
-            $this->error["code"]    = "BadConnection";
-            $this->error["message"] = curl_error($handle);
-            }
-
-        curl_close($handle);
-
-        // return the json
-        return $output;
-        }
+class Geocoder
```

Comment:
Can you direct me to what's really changed here? Looks like some formatting changes are causing a lot to be flagged as new

Reply:
Yeah, Github is funny when displaying the diff for this file, not sure why 🤷‍♂️. The diff looks much better in VSCode (+ GitLens). But the main change here would be the addition of the following methods:

```
  public function isPossible3wa($input);
  public function findPossible3wa($input);
  public function isValid3wa($input);
```

Also renamed a couple of variables to pascalCase, as well as added access specifiers for its properties.